teambit.bvm/install/installWithPnpm.ts

```diff
@@ -0,0 +1,79 @@
+import fs from 'fs';
+import path from 'path';
+import { getConfig, type Config } from '@pnpm/config';
+import { streamParser } from '@pnpm/logger';
+import { initDefaultReporter } from '@pnpm/default-reporter';
+import { install } from '@pnpm/plugin-commands-installation';
+import { readWantedLockfile } from '@pnpm/lockfile.fs';
+
+export async function installWithPnpm(fetch, version: string, dest: string) {
+  await fetchLockfile(fetch, version, dest);
+  await createPackageJsonFile(dest);
+
+  const cliOptions = {
+    argv: [],
+    dir: dest,
+    registry: 'https://node-registry.bit.cloud/',
+  }
+  const { config } = await getConfig({
+    cliOptions,
+    packageManager: { name: '@teambit/bvm', version: '' },
+  });
+  const stopReporting = initReporter(config);
+  await install.handler({
+    ...config,
+    argv: { original: [] },
+    frozenLockfile: true,
+    nodeLinker: 'hoisted',
+    cliOptions,
+    ignoreScripts: true,
+  });
+  // pnpm is doing some actions in workers.
+  // We need to finish them, when we're done.
+  await global['finishWorkers']();
+  stopReporting();
+}
+
+async function fetchLockfile(fetch, version: string, dest: string): Promise<void> {
+  const url = `https://bvm.bit.dev/bit/versions/${version}/pnpm-lock.yaml`;
```

Comment:
allow me to provide the lock file with the --file in bvm install/update (like I can provide you path to the tar file, I want to give you a path to lock file)